jax/_src/random.py

```diff
@@ -72,7 +72,16 @@ def _check_prng_key(key) -> tuple[prng.PRNGKeyArray, bool]:
   if isinstance(key, prng.PRNGKeyArray):
     return key, False
   elif _arraylike(key):
-    if config.jax_enable_custom_prng:
+    if config.jax_raw_prng_key == 'error':
+      raise ValueError(
+        'Raw array passed as key to jax.random function. If this was intended, '
```

Comment:
```suggestion
        'uint32 array passed as key to jax.random function. If this was intended, '
```

Reply:
If we do say "uint32" here, should we *check* the dtype to ensure it's actually u32 before issuing this message? I hope we err on non-u32 inputs at some point already, so that this wouldn't be a change in error behavior...

Reply:
We error on non-u32 inputs in `prng.random_wrap()`, which is called below these statements.

What if I say "Legacy key array" rather than "Raw array" here, and add some more context about using `jax.random.key()`?

Reply:
Since we error here under the `'error'` option, we'd mask the underlying `random_wrap` error. So let's validate here upfront, then proceed to respect the new flag?

Reply:
Done - also added a test to confirm that the key correctness error takes precedence.